### PR TITLE
MOTOR-1163: remove AIOHTTP_NO_EXTENSIONS and test support against Python 3.12

### PR DESCRIPTION
**Pull request by Jibola**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -61,11 +61,9 @@ passenv =
     FLE_AZURE_CLIENTSECRET
     FLE_GCP_EMAIL
     FLE_GCP_PRIVATEKEY
-    AIOHTTP_NO_EXTENSIONS
 
 [testenv:test]
 # TODO: Remove as part of MOTOR-1163
```
Comment (ShaneHarvey) [inline tox.ini:66]: We can remove this todo too.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Could you schedule the 3.12 aiohttp tests?

**Comment by Jibola** [comment]
> Could you schedule the 3.12 aiohttp tests?

Hrm, is there a specific task I needto schedule. I checked evergreen and I see the 3.12 tests already running, as well as the one from the github actions.

https://spruce.mongodb.com/version/65bd41f4a4cf474e8b082ad9/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&variant=%5Emain__ssl~ssl_tox-env~test-py312_os~rhel84%24

**Comment by blink1073** [comment]
aiohttp used to be its own test target, it now runs as part of the standard test suite.

**Review by blink1073 — APPROVED** [review]
LGTM!

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**